molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java

```diff
@@ -138,17 +140,38 @@ public void deleteEntityMeta(String entityName)
 			}
 			entityMetaDataRepository.delete(entityName);
 			if (dataService.hasRepository(entityName)) dataService.removeRepository(entityName);
+			deleteEntityPermissions(entityName);
+
 			return null;
 		});
 
 		refreshCaches();
 	}
 
+	private void deleteEntityPermissions(String entityName)
+	{
+		List<String> authorities = SecurityUtils.getEntityAuthorities(entityName);
+
+		// User permissions
+		if (dataService.hasRepository("UserAuthority"))
+		{
+			Iterable<Entity> userPermissions = dataService.query("UserAuthority").in("role", authorities).findAll();
+			dataService.delete("UserAuthority", userPermissions);
+		}
+
+		// Group permissions
+		if (dataService.hasRepository("GroupAuthority"))
+		{
+			Iterable<Entity> groupPermissions = dataService.query("GroupAuthority").in("role", authorities).findAll();
```

Comment:
In principle yes.
But here we get dependency problems.
That is also the reason that the method is not added to the PermissionManagerService